### DISCUSSION TITLE
How can I set up automatic indentation style detection?

### DISCUSSION THREAD

**Discussion by 8ullyMaguire** [NvChad/NvChad#2536] [Q&A]
I want to automatically detect and use the same indentation style as the current buffer is already using. This is the default behavior in vscode, I think it should be the default behavior in NvChad too.

I've tried adding the [guess-indent plugin](https://github.com/NMAC427/guess-indent.nvim) and restarting but it isn't working. I've also tried using `:GuessIndent` but I get `E492: Not an editor command: GuessIndent`.

`~/.config/nvim/lua/custom/chadrc.lua`
```
local M = {}

M.plugins = "custom.plugins"

return M
```

`~/.config/nvim/lua/custom/plugins.lua`
```
local plugins = {
  {
    "NMAC427/guess-indent.nvim",
    config = function()
      require('guess-indent').setup {}
    end,
    lazy = false
  }
}

return plugins
```

**Comment by siduck** [comment]
https://github.com/NvChad/NvChad/assets/59060246/8b12e1f7-9e85-46f6-88ed-01984f19dee9


works here, also please remove lazy=false and lazyload this with cmd  = "GuessIndent"
